src/dashboard/Data/Browser/AttachSelectedRowsDialog.react.js

```diff
@@ -0,0 +1,169 @@
+
+  componentWillMount() {
+    const { selection, classes, onSelectClass } = this.props;
+    if (selection) {
+      const currentClass = classes[0];
+      const objectIds = [];
+      for (const objectId in selection) {
+        objectIds.push(objectId);
+      }
+      const touchableColumns = onSelectClass(currentClass);
+      const currentColumn = touchableColumns[0];
+      this.setState({
+        currentClass: currentClass,
+        touchableColumns,
+        currentColumn,
+        objectIds,
+      });
+    }
+  }
+
+  handleTargetObjectIdChange(targetObjectId) {
+    this.setState({ targetObjectId });
+  }
+
+  handleConfirm() {
+    const {
+      currentClass,
+      currentColumn,
+      targetObjectId,
+      objectIds,
+    } = this.state;
+    const promise = new Parse.Promise();
+    this.props.onConfirm(currentClass, targetObjectId, currentColumn, objectIds)
+    .then(() => promise.resolve())
+    .catch((error) => promise.reject(error));
+    return promise;
+  }
+
+  handleClassChange(className) {
+    const touchableColumns = this.props.onSelectClass(className);
+    this.setState({
+      currentClass: className,
+      touchableColumns,
+      currentColumn: touchableColumns[0],
+    });
+  }
+
+  handleColumnChange(column) {
+    this.setState({
+      currentColumn: column,
+    });
+  }
+
+  render() {
+    const {
+      relation,
+      onCancel,
+      classes,
+    } = this.props;
+    let targetRelationSelector;
+    let targetEntityIdInsert;
+      if (this.state.touchableColumns.length) {
+        targetRelationSelector = (
+          <Field
+            label={
+              <Label
+                text="Target Relation"
+                description="Target class's relation column"
+              />
+            }
+            input={
+              <Dropdown
+                value={this.state.currentColumn}
+                onChange={this.handleColumnChange}
+              >
+                {this.state.touchableColumns.map(column => (
+                  <Option key={column} value={column}>
+                    {column}
+                  </Option>
+                ))}
+              </Dropdown>
+            }
+          />
+        );
+      }
+      if (this.state.currentColumn) {
+        targetEntityIdInsert = (
+          <Field
+            label={
+              <Label
+                text="Target objectId"
+                description={`${this.state.currentClass} objectId`}
+              />
+            }
+            input={
+              <TextInput
+                placeholder="ox0QZFl7eg, qs81Q72lTL, etc..."
+                value={this.state.targetObjectId}
+                onChange={this.handleTargetObjectIdChange}
+              />
+            }
+          />
+        );
+      }
+    return (
+      <FormModal
+        open
+        icon="plus"
+        iconSize={40}
+        title="Attach Selected Rows to Relation"
+        subtitle="Attach selected rows to specify object's relation"
```

Comment:
I don't think this subtitle really adds anything useful that isn't already in the title, we can probably remove it.